ethsigner/src/main/java/tech/pegasys/ethsigner/requesthandler/sendtransaction/TransactionTransmitter.java

```diff
@@ -0,0 +1,162 @@
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.json.Json;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.web3j.crypto.RawTransaction;
+
+public class TransactionTransmitter {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TransactionTransmitter.class);
+  private static final String JSON_RPC_VERSION = "2.0";
+  private static final String JSON_RPC_METHOD = "eth_sendRawTransaction";
+
+  private final HttpClient ethNodeClient;
+  private final TransactionSerialiser serialiser;
+  private final SendTransactionContext context;
+  private final RetryMechanism retryMechanism;
+  private final HttpResponseFactory responder;
+
+  public TransactionTransmitter(
+      final HttpClient ethNodeClient,
+      final SendTransactionContext context,
+      final TransactionSerialiser serialiser,
+      final RetryMechanism retryMechanism,
+      HttpResponseFactory responder) {
+    this.ethNodeClient = ethNodeClient;
+    this.context = context;
+    this.serialiser = serialiser;
+    this.retryMechanism = retryMechanism;
+    this.responder = responder;
+  }
+
+  public void send() {
+    final JsonRpcBody body = getBody();
+    if (body.hasError()) {
+      reportError(JsonRpcError.INTERNAL_ERROR);
+    } else {
+      sendTransaction(body.body());
+    }
+  }
+
+  private JsonRpcBody getBody() {
+    // This assumes the parameters have already been validated and are correct for the unlocked
+    // account.
+    final String signedTransactionHexString;
+    try {
+      final RawTransaction rawTransaction = context.getRawTransactionBuilder().build();
+      signedTransactionHexString = serialiser.serialise(rawTransaction);
+    } catch (final IllegalArgumentException e) {
+      LOG.debug("Failed to encode transaction: {}", context.getRawTransactionBuilder(), e);
+      return new JsonRpcBody(JsonRpcError.INVALID_PARAMS);
+    } catch (final Throwable e) {
+      LOG.debug(
+          "Failed to encode/serialise transaction: {}", context.getRawTransactionBuilder(), e);
+      return new JsonRpcBody(JsonRpcError.INTERNAL_ERROR);
+    }
+
+    final JsonRpcRequest sendRawTransaction =
+        new JsonRpcRequest(
+            JSON_RPC_VERSION, JSON_RPC_METHOD, singletonList(signedTransactionHexString));
+    sendRawTransaction.setId(context.getId());
+
+    try {
+      return new JsonRpcBody(Json.encodeToBuffer(sendRawTransaction));
+    } catch (final IllegalArgumentException e) {
+      LOG.debug("JSON Serialisation failed for: {}", sendRawTransaction, e);
+      return new JsonRpcBody(JsonRpcError.INTERNAL_ERROR);
+    }
+  }
+
+  private void sendTransaction(final Buffer bodyContent) {
+    final HttpServerRequest httpServerRequest = context.getInitialRequest();
+    final HttpClientRequest proxyRequest =
```

Comment:
this is not really a proxy request, maybe just request?

Reply:
agreed - Done.